### PR TITLE
Fix private decoding bug

### PR DESCRIPTION
**Pull request by BlackHoleFox**
Fixes a bug where the decode implementation of `PrivateKey` would attempt to validate the full key matched itself even when it was encrypted. This is obviously wrong so don't do that. Regression test included that fails when the fix is removed.

~~Depends on https://github.com/badboy/signify-rs/pull/60~~
Bug originally reported in https://github.com/badboy/signify-rs/pull/59#discussion_r1269634345

### CODE REVIEW

File: libsignify/Cargo.toml [hunk 1]
```diff
@@ -17,13 +17,12 @@ repository = "https://github.com/badboy/signify-rs"
 std = []
 
 [dependencies]
-bcrypt-pbkdf = { version = "0.7", default-features = false }
-base64 = { version = "0.13", default-features = false, features = ["alloc"] }
-ed25519-dalek = { version = "1", default-features = false, features = ["alloc", "u64_backend"] }
-rand_core = { version = "0.5", default-features = false }
+bcrypt-pbkdf = { version = "0.10", default-features = false }
+base64ct = { version = "1.6", default-features = false, features = ["alloc"] }
+ed25519-dalek = { version = "2", default-features = false, features = ["alloc", "fast", "rand_core"] }
```
Comment (sug0) [inline libsignify/Cargo.toml]: good call! https://github.com/advisories/GHSA-w5vr-6qhr-36cc

**Review by sug0 — COMMENTED** [review]
I tested these changes, and unfortunately decoding encrypted keys is still broken :(
**EDIT:** well I guess you didn't port every change from the other pr haha

**Comment by BlackHoleFox** [comment]
> I tested these changes, and unfortunately decoding encrypted keys is still broken :(

I don't quite know how to reproduce your issue yet, so no deep dive has been attempted. Is therea way you could create an encrypted key that reproduces the issue which can be uploaded here for testing?

> EDIT: well I guess you didn't port every change from the other pr haha

@sug0 Apologies if that wasn't clear. I had posted on a specific thread in your PR as a scope attempt.

**Comment by sug0** [comment]
@BlackHoleFox of course. clone https://github.com/sug0/git-signify and checkout the branch `signify-decoding-bug`. build the bin (or run `cargo run -- ...args`). under the `keys/` directory you'll find some sample keys I've generated with the `signify` cli util from openbsd; the password to the private key is `1234`.

this util signs arbitrary git objects. let's say we want to sign `HEAD`. we can run:
```
cargo run -- sign -k keys/key.sec HEAD
```
even after you input the right password, you'll never be able to decrypt the private key. if you switch back to `main`, and rebuild the util, you'll be able to decrypt the key and sign `HEAD`.

**Review by badboy — APPROVED** [review]

**Pull request merged**